src/main/java/com/mraof/minestuck/client/gui/ComputerScreen.java

```diff
@@ -14,135 +15,193 @@
+		blit(poseStack, xOffset, yOffset, 0, 0, xSize, ySize);
 		
 		if(!bsod)
 		{
-			if(program != null)
-				program.paintGui(poseStack, this, be);
-			else
-				font.draw(poseStack, "Insert disk.", (width - xSize) / 2F + 15, (height - ySize) / 2F + 45, 4210752);
+			//desktop background
+			RenderSystem.setShaderTexture(0, this.be.getTheme().getTexture());
+			blit(poseStack, xOffset + 9, yOffset + 38, 0, 0, 158, 120);
+			
+			//program and widgets
+			if(program != null) program.paintGui(poseStack, this, be);
+			super.render(poseStack, mouseX, mouseY, partialTicks);
+			
+			//corner bits (goes on top of computer screen slightly)
+			RenderSystem.setShaderTexture(0, guiBackground);
+			blit(poseStack, xOffset + 9, yOffset + 38, xSize, 0, 3, 3);
+			blit(poseStack, xOffset + 164, yOffset + 38, xSize + 3, 0, 3, 3);
+		} else {
+			RenderSystem.setShaderTexture(0, guiBsod);
+			blit(poseStack, xOffset+9, yOffset+38, 0, 0, 158, 120);
 		}
-		
-		super.render(poseStack, mouseX, mouseY, partialTicks);
 	}
 	
-	@Override
-	public boolean isPauseScreen()
+	public void updateGui()
 	{
-		return false;
+		if(program!=null) program.onUpdateGui(this);
 	}
 	
-	@Override
-	public void init()
+	protected void genIcons()
 	{
-		super.init();
+		var xOffset = (width-xSize)/2;
+		var yOffset = (height-ySize)/2;
 		
-		if(be.programSelected == -1 && !be.isBroken())
-			for(Entry<Integer, Boolean> entry : be.installedPrograms.entrySet())
-				if(entry.getValue() && (be.programSelected == -1 || be.programSelected > entry.getKey()))
-					be.programSelected = entry.getKey();
+		int programCount = be.installedPrograms.size();
+		for(int id : be.installedPrograms.stream().sorted().toList())
+		{
+			icons.add(addRenderableWidget(new ComputerIcon(
+					xOffset + 15 + Math.floorDiv(programCount, 5) * 20,
+					yOffset + 24 + programCount % 5 * 20, id)
+			));
+			programCount--;
+		}
+	}
+	
+	protected void setProgram(int id)
+	{
+		if(be.isBroken() || id == -1 || id == -2)
+			return;
+		
+		program = ComputerProgram.getProgram(id);
+		if(program==null) return;
 		
-		if(be.programSelected != -1 && (program == null || program.getId() != be.programSelected))
-			program = ComputerProgram.getProgram(be.programSelected);
+		be.programSelected = id;
+		program.onInitGui(this);
 		
-		programButton = new ExtendedButton((width - xSize) / 2 + 95, (height - ySize) / 2 + 10, 70, 20, Component.empty(), button -> changeProgram());
-		addRenderableWidget(programButton);
-		if(be.programSelected != -1)
-			program.onInitGui(this);
+		for(ComputerIcon icon : icons)
+			icon.visible = false;
 		
 		updateGui();
 	}
 	
-	public void updateGui()
+	protected void exitProgram()
 	{
+		program = null;
+		be.programSelected = -1;
 		
-		programButton.active = be.installedPrograms.size() > 1;
-		
-		if(be.isBroken())
-		{
-			clearWidgets();
-			return;
-		}
+		//TODO find way to make this only clear program widgets so we don't regen icons every time
```

Comment:
After clearing widgets, I suppose you could just add back all icons from the `icons` list: `icons.forEach(this::addRenderableWidget);`